packages/engine/Source/Core/sampleTerrain.js

```diff
@@ -35,25 +36,35 @@ import defined from "./defined.js";
  * // positions[0].height and positions[1].height have been updated.
  * // updatedPositions is just a reference to positions.
  */
-async function sampleTerrain(terrainProvider, level, positions) {
+async function sampleTerrain(
+  terrainProvider,
+  level,
+  positions,
+  rejectOnTileFail
+) {
+  if (!defined(rejectOnTileFail)) {
+    rejectOnTileFail = false;
+  }
   //>>includeStart('debug', pragmas.debug);
   Check.typeOf.object("terrainProvider", terrainProvider);
   Check.typeOf.number("level", level);
+  Check.typeOf.bool("rejectOnTileFail", rejectOnTileFail);
   Check.defined("positions", positions);
   //>>includeEnd('debug');
 
-  return doSampling(terrainProvider, level, positions);
+  return doSampling(terrainProvider, level, positions, rejectOnTileFail);
```

Comment:
* Should this change be reflected in `sampleTerrainMostDetailed` as well?
* Since promises and error handling tends to be a sticking point, I would suggest adding an example to the doc which shows how to use this parameter with a `try/catch` block.